### ISSUE TITLE
Compatibility

### ISSUE THREAD

**Issue by gopinadh39** [wermerb/ngx-mqtt-client#3]
Is It compatible for angular 6

**Comment by wermerb** [comment]
Hi, 
Not yet but i will publish the new version later today.

**Comment by wermerb** [comment]
You can use it if you install `npm i rxjs-compat@6`. I'm working on the fully compatible version.

**Comment by williamBurgeson** [comment]
I ran `ng new mqtt-sample`, added rxjs-compat, mqtt and ngx-mqtt-client, and when I add the module code to app.module 

```
NgxMqttClientModule.withOptions({
        host: 'broker.hivemq.com',
        protocol: 'ws',
        port: 8000,
        path: '/mqtt',
        keepalive: 5
    })
``` 
and run ng serve -o I get the following runtime error in the console:

```
Uncaught ReferenceError: process is not defined
    at Object../node_modules/process-nextick-args/index.js (index.js:3)
    at __webpack_require__ (bootstrap:76)
    at Object../node_modules/readable-stream/lib/_stream_readable.js (_stream_readable.js:26)
    at __webpack_require__ (bootstrap:76)
    at Object../node_modules/readable-stream/readable-browser.js (readable-browser.js:1)
    at __webpack_require__ (bootstrap:76)
    at Object../node_modules/mqtt/lib/store.js (store.js:8)
    at __webpack_require__ (bootstrap:76)
    at Object../node_modules/mqtt/lib/client.js (client.js:7)
    at __webpack_require__ (bootstrap:76)
```
My angular CLI version is 6.0.8 and I'm running the project on a Mac.

I downloaded the project from github and ran without any any obvious problems - looking at the peerdependencies section in that project the only one which doesn't come out of the box with ng cli would be mqtt (which I've added), so I can't think of anything else.

**Comment by wermerb** [comment]
@williamBurgeson 

`process` is node dependency in mqtt.js library. Unfortunately with the new cli they removed node-sims and that causes the issue. I'm a little bit overwhelmed at the moment, but i'm working on the angular 6 update on this project. Hopefully it will be ready at the end of this week.

**Comment by Sergioeabarcaf** [comment]
> @williamBurgeson
> 
> `process` is node dependency in mqtt.js library. Unfortunately with the new cli they removed node-sims and that causes the issue. I'm a little bit overwhelmed at the moment, but i'm working on the angular 6 update on this project. Hopefully it will be ready at the end of this week.


I have the same mistake as @williamBurgeson 

![image](https://user-images.githubusercontent.com/8609473/48023862-b7473b00-e11d-11e8-9eef-bd2568ebfc6c.png)


![image](https://user-images.githubusercontent.com/8609473/48023786-8cf57d80-e11d-11e8-9b52-5b313cd6500f.png)

**Comment by williamBurgeson** [comment]
Are you planning to make it work with both ng 6 and ng 7?

**Comment by wermerb** [comment]
Unfortunately I don't have time to work on this issue :(, if someone has time to create a pr with  webpack build process for this library that would be a huge amount of help. Then we can upgrade the project to version 6&7.

**Comment by Sergioeabarcaf** [comment]
yes, i will like with ng 6 & 7, but i see the package use ng 1.7.4. then what is the last ng version for compatibility?

thanks, for you answers @wermerb @williamBurgeson 

![image](https://user-images.githubusercontent.com/8609473/48071314-cfb86380-e1b8-11e8-8ad5-6529ea759190.png)

**Comment by williamBurgeson** [comment]
ng cli jumped to version 6 alongside ng 6. Now with ng 7 ng cli is also 7, although the build versions are not synchronised.

@Sergioeabarcaf do you have nvm? I've not got round to using it myself but it (should) allows you to have different versions of npm "globally" on the same machine. Within such a setup it should be possible to "globally" install different versions of ng cli.
I'm not 100% on that as I haven't tested.

**Comment by Sergioeabarcaf** [comment]
yes can! here is good explication.

https://blog.angularindepth.com/angular-5-or-angular-6-yes-please-d71b08b5e59b

**Comment by williamBurgeson** [comment]
Doh! Yes of course ng cli works ok with older versions of a project! I forgot!
I'm not sure how many versions back it will go. 
Even if ng cli 7 (global) works with a ng 5 project, I can't see e.g. ng cli 8 or 9 going back to 5. Angular doesn't seem to be that big on backward compatibility...
